tensorflow-ops/src/TensorFlow/Gradient.hs

```diff
@@ -108,6 +109,15 @@ type GradientCompatible a =
     -- TODO(fmayle): MaxPoolGrad doesn't support Double for some reason.
     (Num a, OneOf '[ Float, Complex Float, Complex Double ] a)
 
+-- | Types that gradients can be taken w.r.t. (namely 'Tensor' and 'Variable').
+class GradientTarget t where
```

Comment:
I agree `targetZeros` doesn't make much sense in a general purpose class since `shape (targetZeros x) == shape x` isn't always true. If there is another use case for getting the Output of a tensor/variable, then maybe moving `targetOutput` to a separate class would be useful. Sticking with `GradientTarget` for now.